### PR TITLE
feat: enable switching season

### LINKED ISSUE #775: feat: enable switching sessions from web
- [ ] Dropdown-list with all public sessions
- [ ] A way to join private sessions

### PR DESCRIPTION
**Pull request by Mikatech**
Pathwar implements seasons, but for now it's not possible in the user interface to change season.
resolve #775

### CODE REVIEW

File: web/src/actions/seasons.js [hunk 1]
```diff
@@ -56,6 +57,23 @@ export const fetchPreferences = seasonID => async dispatch => {
   }
 };
 
+export const setSwitchSeason = seasonID => async dispatch => {
+  try {
+    await postPreferences(seasonID);
+
+    dispatch({
+      type: FETCH_PREFERENCES_SUCCESS,
```
Comment (Mikatech) [inline web/src/actions/seasons.js:65]: the type "FETCH_PREFERENCES_SUCCES" don't mean a lot here
same for "FETCH_PREFERENCES_FAILED" below.

File: web/src/actions/userSession.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ import {
   DELETE_ACCOUNT_FAILED,
   DELETE_ACCOUNT_SUCCESS,
   VALIDATE_COUPON_SUCCESS,
-  VALIDATE_COUPON_FAILED,
+  VALIDATE_COUPON_FAILED, SET_ACTIVE_SEASON, SET_ACTIVE_SEASON_FAILED,
```
Comment (Mikatech) [inline web/src/actions/userSession.js:13]: keep the same style as before

File: web/src/pages/HomePage.js [hunk 3]
```diff
@@ -46,6 +49,7 @@ const HomePage = () => {
   const activeChallenges = useSelector(state => state.seasons.activeChallenges);
   const activeSeason = useSelector(state => state.seasons.activeSeason);
   const teamDetails = useSelector(state => state.seasons.teamInDetail);
+  const allSeasons = useSelector(state => state.seasons.allSeasons);
```
Comment (Mikatech) [inline web/src/pages/HomePage.js:52]: could be delete

File: go/pkg/pwapi/api_season-list.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package pwapi
+
+import (
+	"context"
+)
+
+func (svc *service) SeasonList(ctx context.Context, in *SeasonList_Input) (*SeasonList_Output, error) {
+
```
Comment (Doozers) [inline go/pkg/pwapi/api_season-list.go:8]:
```suggestion
```

Comment (Doozers) [inline go/pkg/pwapi/api_season-list.go:8]:
```suggestion
```

File: go/pkg/pwapi/api_season-list.go [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import (
+	"context"
+
+	"pathwar.land/pathwar/v2/go/pkg/pwdb"
+)
+
+func (svc *service) SeasonList(ctx context.Context, in *SeasonList_Input) (*SeasonList_Output, error) {
+	var ret SeasonList_Output
+	svc.db.Where(pwdb.Season{IsGlobal: true, IsTesting: false, Status: 1}).Find(&ret.Items)
```
Comment (moul) [inline go/pkg/pwapi/api_season-list.go:11]:
```suggestion
	svc.db.Where(pwdb.Season{IsTesting: false, Status: 1}).Find(&ret.Items)
```

`IsGlobal` is limited to the default season, the "global season".

Comment (moul) [inline go/pkg/pwapi/api_season-list.go:11]: checkout this please: https://github.com/pathwar/pathwar/blob/master/go/pkg/pwapi/api_user-get-session.go#L56

Comment (Mikatech) [reply go/pkg/pwapi/api_season-list.go]: Thanks, i implemented this function for SeasonList

**Pull request merged**